src/traffic_server/InkAPI.cc

```diff
@@ -4933,6 +4933,43 @@ TSHttpTxnEffectiveUrlStringGet(TSHttpTxn txnp, int *length)
   return sm->t_state.hdr_info.client_request.url_string_get(nullptr, length);
 }
 
+TSReturnCode
+TSHttpHdrEffectiveUrlBuffGet(TSMBuffer hdr_buf, TSMLoc hdr_loc, void *buff, int64_t size, int64_t *length)
```

Comment:
Shouldn't buff be a char*, then you wouldn't have to cast it?  Also, most APIs use buf instead of buff.

Reply:
Those were chosen by @SolidWallOfCode

Reply:
I don't think those are correct and should be changed.

Reply:
What's a good venue for the Bryan - @SolidWallOfCode cage match?  Maybe Comcast in Denver is a good neutral location?

Reply:
"buf" is fine. I like `void * because it makes passing `char *` or `unsigned char *` or `uint8_t *` easier for the caller. Better to have one cast in one place than scattered about the code.

Reply:
Having a void* is a bad idea for a few reasons.

1. It makes it more difficult for a user of the API to understand that a string is being passed back from the API call.
2. The existing API passes back a char* as the URL
3. It makes the API fragile since you can pass it a &int into the API without any type checking.  Having the API use a char* would make the user of the API do a cast and would hopefully realize they are doing something wrong. (last point was from @zwoop)